File: community/cypher/cypher-compiler-2.0/src/main/scala/org/neo4j/cypher/internal/compiler/v2_0/blackbuck/impl/operator/AbstractOperator.java
package org.neo4j.cypher.internal.compiler.v2_0.blackbuck.impl.operator;

import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.sched.Operator;
import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.slot.Cursor;

public abstract class AbstractOperator<C extends Cursor<C>> implements Operator<C>
{
    private final Object key;

    public AbstractOperator( Object key ) {
        this.key = key;
    }

    @Override
    public Object key() {
        return key;
    }
}

File: community/cypher/cypher-compiler-2.0/src/main/scala/org/neo4j/cypher/internal/compiler/v2_0/blackbuck/impl/operator/AbstractOperator.scala
package org.neo4j.cypher.internal.compiler.v2_0.blackbuck.impl.operator

import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.slot.Pos
import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.rows.{RowPool, Cursor}
import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.sched.{Scheduler, Activation, Operator}

abstract class AbstractOperator[P <: Pos[P], C <: Cursor[P, C], E](val key: AnyRef) extends Operator[P, C, E] {
  override def newActivation(pool: RowPool[P, C], scheduler: Scheduler[P, C, E]): AbstractActivation[P, C, E]
}
